kong/plugins/log-serializers/basic.lua

```diff
@@ -5,8 +5,12 @@ function _M.serialize(ngx)
   if ngx.ctx.authenticated_credential ~= nil then
     authenticated_entity = {
       id = ngx.ctx.authenticated_credential.id,
-      consumer_id = ngx.ctx.authenticated_credential.consumer_id
+      consumer_id = ngx.ctx.authenticated_credential.consumer_id,
     }
+    
+    if ngx.ctx.authenticated_consumer ~= nil then
+      authenticated_entity.consumer = ngx.ctx.authenticated_consumer
```

Comment:
unless `ngx.ctx.authenticated_consumer` can actually have a boolean value `false`, this line should go in the table constructor above.

By then that constructor will have 2 accesses to the 'magic' `ngx.ctx` table, which warrants a local caching variable imho.

That said; the whole of this `function` could use some local caching.